packages/next/build/utils.ts

```diff
@@ -1120,6 +1120,17 @@ export function getRawPageExtensions(pageExtensions: string[]): string[] {
   )
 }
 
+export function isPage(
+  filename: string,
+  pagesDir: string,
+  pageExtensions: string[]
+): boolean {
+  return (
+    filename.startsWith(pagesDir) &&
+    new RegExp(`\\.(?:${pageExtensions.join('|')})$`).test(filename)
```

Comment:
Do we need to convert `pageExtensions` to `rawPageExtensions`?

Reply:
@ijjk suggestions?

Reply:
@huozhi, @ijjk, @padmaia, @shuding @timneutkens any suggestions please?

Reply:
```suggestion
    (!pageExtensions.length ||
      pageExtensions.some((ext) => filename.endsWith(ext)))
```

Let's not use a RegExp. It's slightly easier to read IMO.